src/app/3d/qgs3dmaptoolmeasureline.cpp

```diff
@@ -106,12 +115,15 @@ void Qgs3DMapToolMeasureLine::activate()
 void Qgs3DMapToolMeasureLine::deactivate()
 {
   restart();
-  if ( QgsTerrainEntity *terrainEntity = mCanvas->scene()->terrainEntity() )
+  if ( !mCanvasClosed )
```

Comment:
These changes also should not be necessary if the tool would be deactivated before the scene is deleted.